### PR TITLE
[assets] Smaller bundles and remove React dev code.

### PR DESCRIPTION
**Pull request by alloy**
Fixes https://github.com/artsy/reaction-force/issues/154

We were including React dev code in our production bundles (as can be checked with the React Dev Tools installed, as can be seen in the screenshot in the above linked ticket).

Following [the steps outlined in the React docs](https://facebook.github.io/react/docs/optimizing-performance.html#browserify) lead to the following overall reduction in bundle size (and increase in build time):

## Before

```
$ time yarn assets
       50.45 real        58.45 user         8.01 sys

$ du -sh public/assets/
9.8M	public/assets/

$ ls -lSh public/assets/*.js
-rw-r--r--  1 eloy  staff   3.7M May 10 15:48 public/assets/common.js
-rw-r--r--  1 eloy  staff   770K May 10 15:48 public/assets/categories.js
-rw-r--r--  1 eloy  staff   738K May 10 15:48 public/assets/artists_artworks.js
-rw-r--r--  1 eloy  staff   548K May 10 15:48 public/assets/misc.js
-rw-r--r--  1 eloy  staff   406K May 10 15:47 public/assets/auctions.js
-rw-r--r--  1 eloy  staff   374K May 10 15:47 public/assets/partners.js
-rw-r--r--  1 eloy  staff   365K May 10 15:47 public/assets/mobile_all.js
-rw-r--r--  1 eloy  staff   105K May 10 15:47 public/assets/fairs.js
-rw-r--r--  1 eloy  staff    87K May 10 15:47 public/assets/analytics.js
-rw-r--r--  1 eloy  staff    58K May 10 15:47 public/assets/editorial_features.js
-rw-r--r--  1 eloy  staff    54K May 10 15:47 public/assets/home.js
-rw-r--r--  1 eloy  staff    47K May 10 15:47 public/assets/mobile_analytics.js
-rw-r--r--  1 eloy  staff    31K May 10 15:47 public/assets/mobile_articles.js
-rw-r--r--  1 eloy  staff    26K May 10 15:47 public/assets/mobile_notifications.js
-rw-r--r--  1 eloy  staff    15K May 10 15:47 public/assets/main_layout.js
-rw-r--r--  1 eloy  staff    11K May 10 15:47 public/assets/mobile_show.js
-rw-r--r--  1 eloy  staff   9.2K May 10 15:47 public/assets/article.js
-rw-r--r--  1 eloy  staff   8.9K May 10 15:47 public/assets/mobile_artwork.js
-rw-r--r--  1 eloy  staff   5.5K May 10 15:47 public/assets/articles.js
-rw-r--r--  1 eloy  staff   4.1K May 10 15:47 public/assets/mobile_favorites_following.js
-rw-r--r--  1 eloy  staff   3.7K May 10 15:47 public/assets/json_page.js
-rw-r--r--  1 eloy  staff   2.4K May 10 15:47 public/assets/mobile_user.js
-rw-r--r--  1 eloy  staff   1.7K May 10 15:47 public/assets/mobile_artists.js
-rw-r--r--  1 eloy  staff   1.6K May 10 15:47 public/assets/eoy_2016.js
-rw-r--r--  1 eloy  staff   550B May 10 15:47 public/assets/mobile_openseadragon.js
-rw-r--r--  1 eloy  staff   506B May 10 15:47 public/assets/collect_art.js
```

## After

```
$ time yarn assets
      135.58 real       137.72 user        10.26 sys

$ du -sh public/assets/
7.9M	public/assets/

$ ls -lSh public/assets/*.js
-rw-r--r--  1 eloy  staff   2.7M May 10 15:38 public/assets/common.js
-rw-r--r--  1 eloy  staff   563K May 10 15:37 public/assets/artists_artworks.js
-rw-r--r--  1 eloy  staff   539K May 10 15:37 public/assets/categories.js
-rw-r--r--  1 eloy  staff   411K May 10 15:37 public/assets/misc.js
-rw-r--r--  1 eloy  staff   293K May 10 15:37 public/assets/auctions.js
-rw-r--r--  1 eloy  staff   278K May 10 15:37 public/assets/partners.js
-rw-r--r--  1 eloy  staff   266K May 10 15:37 public/assets/mobile_all.js
-rw-r--r--  1 eloy  staff    79K May 10 15:37 public/assets/fairs.js
-rw-r--r--  1 eloy  staff    69K May 10 15:37 public/assets/analytics.js
-rw-r--r--  1 eloy  staff    48K May 10 15:37 public/assets/editorial_features.js
-rw-r--r--  1 eloy  staff    41K May 10 15:37 public/assets/home.js
-rw-r--r--  1 eloy  staff    36K May 10 15:37 public/assets/mobile_analytics.js
-rw-r--r--  1 eloy  staff    24K May 10 15:37 public/assets/mobile_articles.js
-rw-r--r--  1 eloy  staff    17K May 10 15:37 public/assets/mobile_notifications.js
-rw-r--r--  1 eloy  staff    12K May 10 15:37 public/assets/main_layout.js
-rw-r--r--  1 eloy  staff   8.1K May 10 15:37 public/assets/mobile_show.js
-rw-r--r--  1 eloy  staff   7.0K May 10 15:37 public/assets/article.js
-rw-r--r--  1 eloy  staff   6.9K May 10 15:37 public/assets/mobile_artwork.js
-rw-r--r--  1 eloy  staff   3.8K May 10 15:37 public/assets/articles.js
-rw-r--r--  1 eloy  staff   3.1K May 10 15:37 public/assets/json_page.js
-rw-r--r--  1 eloy  staff   3.0K May 10 15:37 public/assets/mobile_favorites_following.js
-rw-r--r--  1 eloy  staff   1.9K May 10 15:37 public/assets/mobile_user.js
-rw-r--r--  1 eloy  staff   1.4K May 10 15:37 public/assets/mobile_artists.js
-rw-r--r--  1 eloy  staff   1.3K May 10 15:37 public/assets/eoy_2016.js
-rw-r--r--  1 eloy  staff   530B May 10 15:37 public/assets/mobile_openseadragon.js
-rw-r--r--  1 eloy  staff   506B May 10 15:37 public/assets/collect_art.js
```

----

@craigspaeth I’ve done some cursory testing of this build locally, but having not dealt with browserify at all yet I’d like your expert opinion and also think this probably requires everybody to do some poper testing in staging?

### CODE REVIEW

**Comment by alloy** [comment]
Offtopic: but I can’t find any source maps for these bundles, where/how is that being handled?

File: package.json [hunk 1]
```diff
@@ -173,6 +175,7 @@
     "stickyfill": "^1.1.1",
     "supertest": "^2.0.1",
     "typeahead.js": "^0.10.5",
+    "uglifyify": "^3.0.4",
```
Comment (alloy) [inline package.json]: I have to agree, not a pretty name.

**Comment by craigspaeth** [comment]
Nice, thanks for endeavoring on this! I see React's docs _do_ suggest envify—sorry about that @damassi I swore it only required `NODE_ENV=production` in the past. Source maps can be enabled in browserify via the `--debug` flag and [extracted via a plugin](https://github.com/substack/node-browserify#external-source-maps).

**Comment by alloy** [comment]
Cool, thanks, I’ll look at source maps from browserify as part of my compiled Force branch 👍

**Comment by damassi** [comment]
@alloy - if you're in there and looking around and see a quick way to toggle livereload on and off after dev compiles that would be so appreciated :)

**Comment by alloy** [comment]
@damassi I don’t quite follow. Could you rephrase?

**Comment by damassi** [comment]
Ah! apologies -- `livereload` is like a pre-HMR technology where after something compiles it will send a reload request to the browser so that devs don't have to manually refresh every time something changes. I've been thinking about adding something like this https://www.npmjs.com/package/browserify-livereload but have been bogged down with other work.

**Comment by damassi** [comment]
(With the work y'all are doing on `reaction-force` this would quickly become irrelevant though.)

**Comment by alloy** [comment]
Oh I see. I’d feel more comfortable making such changes if I were actually using it, to be honest.

Also, if it’s _just_ reloading a window, it kinda sounds like you could just use any ol’ file watcher, no? E.g. using kicker: `kicker -e 'open http://localhost:5000/foo/bar' desktop/apps/auctions`

**Comment by damassi** [comment]
Totally -- was more of a wink than a request 😄  It's something I've been wanting to add for a little bit to make things a bit quicker in Force.
